src/ray/core_worker/task_interface.h

```diff
@@ -30,58 +30,84 @@ struct TaskOptions {
 struct ActorCreationOptions {
   ActorCreationOptions() {}
   ActorCreationOptions(uint64_t max_reconstructions,
-                       const std::unordered_map<std::string, double> &resources)
-      : max_reconstructions(max_reconstructions), resources(resources) {}
+                       const std::unordered_map<std::string, double> &resources,
+                       const std::vector<std::string> &dynamic_worker_options)
+      : max_reconstructions(max_reconstructions),
+        resources(resources),
+        dynamic_worker_options(dynamic_worker_options) {}
 
   /// Maximum number of times that the actor should be reconstructed when it dies
   /// unexpectedly. It must be non-negative. If it's 0, the actor won't be reconstructed.
   const uint64_t max_reconstructions = 0;
   /// Resources required by the whole lifetime of this actor.
   const std::unordered_map<std::string, double> resources;
+  /// The dynamic options used in the worker command when starting the worker process for
+  /// an actor creation task.
+  const std::vector<std::string> dynamic_worker_options;
 };
 
 /// A handle to an actor.
 class ActorHandle {
  public:
-  ActorHandle(const ActorID &actor_id, const ActorHandleID &actor_handle_id)
-      : actor_id_(actor_id),
-        actor_handle_id_(actor_handle_id),
-        actor_cursor_(ObjectID::FromBinary(actor_id.Binary())),
-        task_counter_(0) {}
+  ActorHandle(const ActorID &actor_id, const ActorHandleID &actor_handle_id,
+              const ray::rpc::Language actor_language,
+              const std::vector<std::string> &actor_creation_task_function_descriptor);
+
+  ActorHandle(const ActorHandle &other);
 
   /// ID of the actor.
-  const ray::ActorID &ActorID() const { return actor_id_; };
+  const ray::ActorID ActorID() const;
 
   /// ID of this actor handle.
-  const ray::ActorHandleID &ActorHandleID() const { return actor_handle_id_; };
+  const ray::ActorHandleID ActorHandleID() const;
+
+  /// Language of the actor.
+  const ray::rpc::Language ActorLanguage() const;
+
+  // Function descriptor of actor creation task.
+  const std::vector<std::string> ActorCreationTaskFunctionDescriptor() const;
+
+  /// The unique id of the last return of the last task.
+  /// It's used as a dependency for the next task.
+  const ObjectID ActorCursor() const;
+
+  /// The number of tasks that have been invoked on this actor.
+  const int64_t TaskCounter() const;
+
+  /// The number of times that this actor handle has been forked.
+  /// It's used to make sure ids of actor handles are unique.
+  const int64_t NumForks() const;
+
+  ActorHandle Fork();
+
+  void Serialize(std::string *output);
+
+  static ActorHandle Deserialize(const std::string &data);
 
  private:
-  /// Cursor of this actor.
-  const ObjectID &ActorCursor() const { return actor_cursor_; };
+  ActorHandle();
```

Comment:
why private instead of just non-existent?

Reply:
We need it to create an empty instance in `Fork` and `Deserialize` function, and then fill fields.